File: 217.contains-duplicate.py
#
# @lc app=leetcode id=217 lang=python3
#
# [217] Contains Duplicate
#

# @lc code=start
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        
        unique_nums = set()

        for n in nums:
            if n not in unique_nums:
                unique_nums.add(n)
            else:
                return True
        
        return False
# @lc code=end

File: 217.contains-duplicate.java
/*
 * @lc app=leetcode id=217 lang=java
 *
 * [217] Contains Duplicate
 */

// @lc code=start

import java.util.HashSet;
import java.util.Set;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> uniques = new HashSet<>();

        for (int n : nums) {
            if (uniques.contains(n)) {
                return true;
            }

            uniques.add(n);
        }

        return false;
    }
}
// @lc code=end

